src/rpc_api/data_types.rs

```diff
@@ -109,6 +106,27 @@ pub struct MarketDeal {
     pub state: DealState,
 }
 
+impl Clone for MarketDeal {
```

Comment:
Add an issue to fil-actor-interface, and a tracking issue in this repo to implement `Clone` for `DealProposal`